src/database/queries.ts

```diff
@@ -82,6 +86,58 @@ export const getVoucher = async (uuid: uuid): Promise<Voucher> => {
   }
 };
 
+// TODO: Convert ENUM to actual type
+export const createVoucher = async (voucher: VoucherCreate): Promise<uuid> => {
+  try {
+    const docRef = await addDoc(voucherCollection, voucher);
+    console.log('Document ref id: ', docRef.id);
+    return docRef.id;
+  } catch (e) {
+    console.warn('(createVoucher)', e);
+    throw e;
+  }
+};
+
+/**
+ * Find a Voucher using uuid and update its vendorUuid if found.
+ */
+export const setVoucherVendorUuid = async (
```

Comment:
Since there's a lot of repeated code between the various queries that update Voucher data, you can try creating a single `updateVoucher` helper function and have your various setters call `updateVoucher`. This way, your setters can handle all the logic related to the specific field you're setting (e.g. if later down the road we want to add logic to validate/manipulate the inputs) and `updateVoucher` is the only place we actually interact with firebase. Here's an outline:

```
const updateVoucher = async (voucher: Partial<Voucher>) => {
  try {
    const docRef = doc(voucherCollection, voucher.uuid);
    await updateDoc(...);
  } catch (e) {
    console.warn('(updateVoucher)', e);
    throw e;
  }
};

export const setVoucherStatus = async (uuid: uuid, status: string) =>
  updateVoucher({ uuid, status });

export const setVoucherVendorUuid = async (uuid: uuid, vendorUuid: uuid) =>
  updateVoucher({ uuid, vendorUuid });
```

Reply:
Notice how `updateVoucher` is not exported (and therefore cannot be used elsewhere in the app), whereas the setters are exported.